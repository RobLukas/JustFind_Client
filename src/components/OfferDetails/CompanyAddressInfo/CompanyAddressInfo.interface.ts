import { SvgIconTypeMap } from '@material-ui/core';
import { OverridableComponent } from '@material-ui/core/OverridableComponent';

interface CompanyAddressInfoProps {
  children: string;
  icon: OverridableComponent<SvgIconTypeMap>;
}

export default CompanyAddressInfoProps;
