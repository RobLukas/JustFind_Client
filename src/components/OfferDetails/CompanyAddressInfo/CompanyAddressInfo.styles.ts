import styled from 'styled-components';
import { SvgIcon } from '@material-ui/core';

const AddressTitle = styled.span(({ theme }) => ({
  verticalAlign: 'middle',
  textAlign: 'center',
  color: theme.styles.secondary.color,
  marginLeft: 3,
  fontSize: 11,
}));

const MuiIcon = styled(SvgIcon)`
  &.MuiSvgIcon-root {
    vertical-align: 'text-top';
    font-size: 0.875rem;
  }
` as typeof SvgIcon;

export default { AddressTitle, MuiIcon };
