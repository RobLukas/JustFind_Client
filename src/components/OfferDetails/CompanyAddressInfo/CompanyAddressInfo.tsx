import React, { FC } from 'react';
import CompanyAddressInfoProps from './CompanyAddressInfo.interface';
import S from './CompanyAddressInfo.styles';

const CompanyAddressInfo: FC<CompanyAddressInfoProps> = ({
  children,
  icon,
}: CompanyAddressInfoProps) => {
  return (
    <>
      <S.MuiIcon component={icon} />
      <S.AddressTitle>{children}</S.AddressTitle>
    </>
  );
};

export default CompanyAddressInfo;
