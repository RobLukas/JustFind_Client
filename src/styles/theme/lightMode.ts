export default {
  name: 'Light Theme',
  styles: {
    primary: {
      background: '#f3f6f8',
      color: '#37474f',
    },
    secondary: {
      background: '#fff',
      color: '#99a1ab',
    },
    dropdown: {
      button: {
        hover: '#f3f6f8',
      },
    },
    scrollbar: {
      thumbBackground: '#c1c1c1',
      background: 'rgba(0,0,0,0.07)',
      hover: 'rgba(0,0,0,0.09)',
    },
    filter: {
      button: {
        color: '#777',
      },
      selected: {
        background: 'linear-gradient(to right, #ba68c8, #9c27b0)',
        color: '#fff',
      },
    },
    offer: {
      header: {
        title: '#fff',
        filter: {
          brightness: '1',
        },
        icons: {
          companyName: '#ff5252',
          companySize: '#fb8c00',
          companyEMPType: '#ab47bc',
          companySkill: '#66bb6a',
          new: '#448aff',
        },
      },
      desc: {
        seperateLine: {
          color: '#f3f6f8',
        },
        skillPoint: {
          selected: '#ff4081',
          unselected: '#393939',
        },
      },
    },
    offers: {
      newTitle: {
        background: '#d9ddfc',
        color: '#4256ef',
      },
      techTitle: {
        background: '#fff',
        border: '#e0e0e0',
      },
    },
  },
};
