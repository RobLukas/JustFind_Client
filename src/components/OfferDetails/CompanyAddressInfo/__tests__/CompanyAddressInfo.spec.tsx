import React from 'react';
import { SvgIcon } from '@material-ui/core';
import { cleanup, render } from '@testing-library/react';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';

import CompanyAddressInfoProps from '../CompanyAddressInfo.interface';
import CompanyAddressInfo from '../CompanyAddressInfo';
import lightMode from 'styles/theme/lightMode';

describe('<CompanyAddressInfo />', () => {
  afterEach(cleanup);
  it('render correctly', () => {
    const { children, icon }: CompanyAddressInfoProps = {
      children: 'address',
      icon: SvgIcon,
    };
    const { container } = render(
      <ThemeProvider theme={lightMode}>
        <CompanyAddressInfo icon={icon}>{children}</CompanyAddressInfo>
      </ThemeProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
