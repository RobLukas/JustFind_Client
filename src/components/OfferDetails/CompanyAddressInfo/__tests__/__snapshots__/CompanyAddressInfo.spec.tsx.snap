// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompanyAddressInfo /> render correctly 1`] = `
<svg
  aria-hidden="true"
  class="MuiSvgIcon-root MuiSvgIcon-root sc-gtssRu lgkXBe"
  focusable="false"
  viewBox="0 0 24 24"
/>
`;
